bootstrap/kubeadm/controllers/kubeadmconfig_controller_test.go

```diff
@@ -152,7 +150,7 @@ func TestKubeadmConfigReconciler_Reconcile_ReturnErrorIfReferencedMachineIsNotFo
 		},
 	}
 	_, err := k.Reconcile(ctx, request)
-	g.Expect(err).To(HaveOccurred())
+	g.Expect(err).To(BeNil())
```

Comment:
How did this change? 🤔

Reply:
This test was incorrect (as noted in the skip comment)
So I have changed it to match the logic implemented in the controller

Reply:
🤷‍♂️ I will make some investigation on the release 0.3 branch

Reply:
@vincepri  the different behavior is due to a change in `ReasonForError` func in apimachinery (https://github.com/kubernetes/kubernetes/commit/3244350046412705939c1f7ad5bd086a5bf73f66)
Before this change, the call to https://github.com/kubernetes-sigs/cluster-api/blob/6404141e933ce31fae6d48841e10475c843a2ba0/bootstrap/kubeadm/controllers/kubeadmconfig_controller.go#L140 was returning false, because we are wrapping the API error.
Now instead the call to `apierrors.IsNotFound` properly recognized the root cause of the error (object not found), and so it returns `nil`.

Reply:
Awesome! Good to know it got fixed!